airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,571 @@
+    def _get_field_name_with_parent(field_name, parent):
+        if parent:
+            return parent + '.' + field_name
+        return field_name
+
+    @staticmethod
+    def _sanity_checks(children_validation_specs, field_type, full_field_path,
+                       regexp, custom_validation, value):
+        # type: (dict, str, str, str, function, object) -> None
+        if value is None and field_type != 'union':
+            raise FieldValidationException(
+                "The required field '{}' is missing".format(full_field_path))
+        if regexp and field_type:
+            raise ValidationSpecificationException(
+                "The field '{}' has both type and regexp defined. The regexp is only"
+                "allowed without type (i.e. assume type is 'str' "
+                "that can be validated with regexp)".format(full_field_path))
+        if children_validation_specs and field_type not in COMPOSITE_FIELD_TYPES:
+            raise ValidationSpecificationException(
+                "Fields are specified in field '{}' which is of type '{}'. "
+                "Fields are only allowed for composite fields ('{}').".
+                format(full_field_path, field_type, COMPOSITE_FIELD_TYPES))
+        if custom_validation and field_type:
+            raise ValidationSpecificationException(
+                "The field '{}' has both type and custom_validation defined. "
+                "Custom validation is only allowed without type so that "
+                "custom validation can check type on its own".
+                format(full_field_path))
+
+    @staticmethod
+    def _validate_regexp(full_field_path, regexp, value):
+        # type: (str, str, str) -> None
+        if not re.match(regexp, value):
+            # Note matching of only the beginning as we assume the regexps all-or-nothing
+            raise FieldValidationException(
+                "The field '{}' value '{}' does not match regexp {}.".
+                format(full_field_path, value, regexp))
+
+    def _validate_dict(self, children_validation_specs, full_field_path, value):
+        # type: (dict, str, dict) -> None
+        for child_validation_spec in children_validation_specs:
+            self._validate_field(validation_spec=child_validation_spec,
+                                 dictionary_to_validate=value,
+                                 parent=full_field_path)
+        # Possibly we find out that we should not validate extra fields
+        # for future compatibility ?
+        # But this is super helpful to detect typos early,
+        # so we might want to leave it here. It's super easy
+        # To add future validations
+        # (and we might want to disable this check with some flag)
+        for field_name in value.keys():
+            if field_name not in [spec['name'] for spec in children_validation_specs]:
+                self.log.warning(
+                    "The field '{}' is present in the dictionary but "
```

Comment:
This error message is very unclear to the end user.

Reply:
Fixed in 3722a2be480c6d8080fc60491e8fa431b7eb6590